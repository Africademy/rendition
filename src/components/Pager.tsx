import * as React from 'react';
import FaChevronLeft = require('react-icons/lib/fa/chevron-left');
import FaChevronRight = require('react-icons/lib/fa/chevron-right');
import { ButtonGroup } from '../';
import Button from './Button';
import { Flex, FlexProps } from './Grid';
import Txt from './Txt';

const Pager = ({
	totalItems,
	itemsPerPage,
	page,
	nextPage,
	prevPage,
	...props
}: PagerProps) => {
	const lowerBound = page * itemsPerPage;
	const upperBound = Math.min((page + 1) * itemsPerPage, totalItems);

	return (
		<Flex justifyContent="flex-end" alignItems="center" {...props}>
			<Txt mr={2}>
				<strong>
					{lowerBound + 1} - {upperBound}
				</strong>{' '}
				of <strong>{totalItems}</strong>
			</Txt>

			<Flex>
				<ButtonGroup>
					<Button
						px={11}
						className="rendition-pager__btn--prev"
						disabled={totalItems <= itemsPerPage || page <= 0}
						onClick={() => prevPage()}
					>
						<FaChevronLeft />
					</Button>
					<Button
						px={11}
						className="rendition-pager__btn--next"
						disabled={
							totalItems <= itemsPerPage ||
							totalItems / itemsPerPage <= page + 1
						}
						onClick={() => nextPage()}
					>
						<FaChevronRight />
					</Button>
				</ButtonGroup>
			</Flex>
		</Flex>
	);
};

export interface PagerProps extends FlexProps {
	totalItems: number;
	itemsPerPage: number;
	page: number;
	nextPage: () => void;
	prevPage: () => void;
}

export default Pager;
