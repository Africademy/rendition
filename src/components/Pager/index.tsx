import * as React from 'react';
import FaChevronLeft from 'react-icons/lib/fa/chevron-left';
import FaChevronRight from 'react-icons/lib/fa/chevron-right';
import Button from '../Button';
import ButtonGroup from '../ButtonGroup';
import { Flex, FlexProps } from '../Grid';
import Txt from '../Txt';

const Pager = ({
	totalItems,
	itemsPerPage,
	page,
	nextPage,
	prevPage,
	...props
}: PagerProps) => {
	const lowerBound = page * itemsPerPage;
	const upperBound = Math.min((page + 1) * itemsPerPage, totalItems);

	return (
		<Flex justifyContent="flex-end" alignItems="center" {...props}>
			<Txt mr={2}>
				<strong>
					{lowerBound + 1} - {upperBound}
				</strong>{' '}
				of <strong>{totalItems}</strong>
			</Txt>

			<Flex>
				<ButtonGroup>
					<Button
						quartenary
						outline
						className="rendition-pager__btn--prev"
						disabled={totalItems <= itemsPerPage || page <= 0}
						onClick={() => prevPage()}
						icon={<FaChevronLeft />}
					/>
					<Button
						quartenary
						outline
						className="rendition-pager__btn--next"
						disabled={
							totalItems <= itemsPerPage ||
							totalItems / itemsPerPage <= page + 1
						}
						onClick={() => nextPage()}
						icon={<FaChevronRight />}
					/>
				</ButtonGroup>
			</Flex>
		</Flex>
	);
};

export interface PagerProps extends FlexProps {
	totalItems: number;
	itemsPerPage: number;
	page: number;
	nextPage: () => void;
	prevPage: () => void;
}

export default Pager;
