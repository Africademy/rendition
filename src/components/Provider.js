import * as React from 'react'
import styled, { ThemeProvider } from 'styled-components'
import defaultTheme from '../theme'

const Base = styled.div`
  font-family: ${props => props.theme.font};
`

const Provider = ({ theme, ...props }) => {
  return (
    <ThemeProvider theme={theme || defaultTheme}>
      <Base {...props} />
    </ThemeProvider>
  )
}

export default Provider
