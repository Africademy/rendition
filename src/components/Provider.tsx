import * as React from 'react';
import styled, { StyledFunction, ThemeProvider } from 'styled-components';
import { DefaultProps, Theme } from '../common-types';
import defaultTheme from '../theme';

export interface ThemedBaseProps extends DefaultProps {
	theme?: Theme;
}

const Base = (styled.div as StyledFunction<DefaultProps>)`
  font-family: ${props => props.theme.font};
`;

const Provider = ({ theme, ...props }: ThemedBaseProps) => {
	return (
		<ThemeProvider theme={theme || defaultTheme}>
			<Base {...props} />
		</ThemeProvider>
	);
};

export default Provider;
