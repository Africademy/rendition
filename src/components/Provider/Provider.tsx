import {
	Grommet
} from 'grommet'
import * as React from 'react';

const Provider = ({ theme, ...props }: any) => {
	return (
		<Grommet theme={theme} {...props} />
	);
};

export default Provider;
