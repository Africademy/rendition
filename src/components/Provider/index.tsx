import { Grommet } from 'grommet';
import merge from 'lodash/merge';
import * as React from 'react';
import styled from 'styled-components';
import { DefaultProps, Theme } from '../../common-types';
import defaultTheme from '../../theme';
import { px } from '../../utils';
import { BreakpointProvider } from './BreakpointProvider';

const Base = styled(Grommet)`
	font-family: ${props => props.theme.font};
	font-size: ${props => px(props.theme.fontSizes[1])};

	h1,
	h2,
	h3,
	h4,
	h5,
	h6,
	button {
		font-family: ${props => props.theme.titleFont};
	}
`;

const Provider = ({ theme, ...props }: ThemedProvider) => {
	const providerTheme = merge(defaultTheme, theme);
	return (
		<BreakpointProvider breakpoints={providerTheme.breakpoints}>
			<Base theme={providerTheme} {...props} />
		</BreakpointProvider>
	);
};

export interface ThemedProvider extends DefaultProps {
	theme?: Partial<Theme>;
}

export default Provider;
