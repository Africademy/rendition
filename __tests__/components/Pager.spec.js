/* globals expect, describe, it */
import { mount } from 'enzyme'
import * as _ from 'lodash'
import React from 'react'
import renderer from 'react-test-renderer'
import sinon from 'sinon'
import Provider from '../../src/components/Provider'
import Pager from '../../src/components/Pager'

describe('Pager component', () => {
  it('should match the stored snapshot', () => {
    const component = renderer.create(
      <Provider>
        <Pager
          totalItems={436}
          itemsPerPage={50}
          page={0}
          nextPage={_.noop}
          prevPage={_.noop}
        />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should disable the "prev" button on page 0', () => {
    const callback = sinon.spy()
    const component = mount(
      <Provider>
        <Pager
          totalItems={436}
          itemsPerPage={50}
          page={0}
          nextPage={_.noop}
          prevPage={callback}
        />
      </Provider>
    )

    const firstButton = component.find('button').first()
    firstButton.simulate('click')

    expect(callback.callCount).toEqual(0)
    expect(firstButton.prop('disabled')).toEqual(true)
  })

  it('should disable the "next" button on the last page', () => {
    const callback = sinon.spy()
    const component = mount(
      <Provider>
        <Pager
          totalItems={100}
          itemsPerPage={50}
          page={1}
          nextPage={callback}
          prevPage={_.noop}
        />
      </Provider>
    )

    const lastButton = component.find('button').last()
    lastButton.simulate('click')

    expect(callback.callCount).toEqual(0)
    expect(lastButton.prop('disabled')).toEqual(true)
  })
})
