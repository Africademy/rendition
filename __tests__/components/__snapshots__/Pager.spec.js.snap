// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager component should match the stored snapshot 1`] = `
<div
  className="sc-bdVaJa iLLxmY"
>
  <div
    className="sc-cSHVUG ddiRBz sc-jzJRlG chuvNI sc-fjdhpX fGySkI"
  >
    <div
      className="sc-chPdSV bErQmz sc-kAzzGY duiWHn"
    >
      <strong>
        1
         - 
        50
      </strong>
       
      of 
      <strong>
        436
      </strong>
    </div>
    <div
      className="sc-cSHVUG dzaSRZ sc-jzJRlG chuvNI sc-fjdhpX fGySkI"
    >
      <div
        className="sc-kpOJdX brCIVB sc-jzJRlG chuvNI sc-fjdhpX fGySkI"
      >
        <button
          className="sc-htoDjs rendition-pager__btn--prev sc-VigVT dzpHnm sc-gqjmRU dTxmFc"
          disabled={true}
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.1 6.7l-11.8 11.9 11.8 11.8q0.5 0.4 0.5 1t-0.5 1l-3.7 3.7q-0.4 0.5-1 0.5t-1-0.5l-16.5-16.5q-0.5-0.4-0.5-1t0.5-1l16.5-16.6q0.5-0.4 1-0.4t1 0.4l3.7 3.7q0.5 0.4 0.5 1t-0.5 1z"
              />
            </g>
          </svg>
        </button>
        <button
          className="sc-htoDjs rendition-pager__btn--next sc-VigVT dzpHnm sc-gqjmRU dTxmFc"
          disabled={false}
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.2 19.6l-16.6 16.5q-0.4 0.5-1 0.5t-1-0.5l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l11.9-11.8-11.9-11.9q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
